cep/source/man/cepInterpUi.3
.\" This manpage has been automatically generated by docbook2man 
.\" from a DocBook document.  This tool can be found at:
.\" <http://shell.ipoline.com/~elmert/comp/docbook2X/> 
.\" Please send any bug reports, improvements, comments, patches, 
.\" etc. to Steve Cheng <steve@ggi-project.org>.
.TH "cepInterpUi" "3" "13 November 2002" "" ""
.SH NAME
cepInterpUi \- Displays Interpolationh user interface
.SH SYNOPSIS

.nf
 The follwing is an example of how to create this object.
 
 cepInterpUi interpUi;
 
.fi
.SH "DESCRIPTION"
.PP
An implementation of the <command>cepInterpUi</command> class
which displays the interpolation GUI.
</para>
<para>
<command>void showSampleRate(double val)</command>
Shows the interpolation GUI.
<para><itemizedlist>
<listitem><para>val:- the default value of the sample rate. This is usually 1/2 the distance
between first two points in the dataset</para></listitem>
</itemizedlist></para>
</para>
<para>
<command> double getSampleRate()</command>
Gets the sample rate selected by the user. If a value of -1 is returned this operation
has been canceled. If -2 is returned the value entered was invalid.
</para>
<para>
<command> cepInterpShowRate::units getUnits()</command>
Returns the units selected by the user. This fuction will return an enumerated type
which is one of:-
<para><itemizedlist>
<listitem><para>Years</para></listitem>
<listitem><para>Days</para></listitem>
<listitem><para>Hours</para></listitem>
<listitem><para>Unknowen</para></listitem>
</itemizedlist></para>
.SH "VERSION"
.PP
This documentation was generated for the \fBGeodetic Data Manipulation System\fR 0.1 by \fBautodocbook\fR (http://www.stillhq.com).
.SH "AUTHOR"
.PP
The \fBGeodetic Data Manipulation System\fR is under development by Michael Still (mikal@stillhq.com), Daniel Fernandez, Kristy Van Der Vlist, Blake Swadling, and Nick Wheatstone. All code is Copyright 2002 Michael Still, Daniel Fernandez, Kristy Van Der Vlist, Blake Swadling and Nick Wheatstone,  and is released under the GNU GPL. Code submissions are welcomed. Contact Michael at the address above if you have something you would like to contribute.
.SH "BUGS"
.PP
There  are no known bugs in the \fBGeodetic Data Manipulation System\fR. If you find one, please contact mikal@stillhq.com and let me know.


cep/ChangeLog
2002-02-27 Michael Still mikal@stillhq.com
	Diarised the meeting of 26 February 2002

2002-03-01 Michael Still mikal@stillhq.com
	Diarised the meeting from today

2002-03-01 Michael Still mikal@stillhq.com
	Started work on the intial report for the project proposal... This was mainly building the skeleton of the document which will be fleshed out at the weekend meeting

2002-04-29 Daniel
	Added cepWindows class
	
2002-05-01 Michael Still mikal@stillhq.com
	Meeting reports

2002-05-01 Michael Still mikal@stillhq.com
	Added cepWindows to build process and fixed compile errors

2002-05-04 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	Fixed some todos in cepWindows.cpp and began implementation of 
	vector for using data and arrow of windows.

2002-05-04 Blake Swadling blake@swadling.org
	Added the source files and ps files for the initial propject proposal.
	Added matlab files for preliminary prototype

2002-05-06 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	Further addressed some todos.
	changed implementation of vectors to suit definition. Not yet 
	complete.

2002-05-07 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	Addressed more todos.
	Fixed implementation of vectors in both windows and ham.

2002-05-07 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	Added constructor to cepDataset.cpp to deal with window and 
	hamming return paramaters.

2002-05-13 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	Added resize functionality to vector in cepWindows.cpp
	Addressed more todos in order add cepWindows.cpp code to cepDataset.cpp
	Added cepWindows.cpp code to cepDataset.cpp	

2002-05-13 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	Added the fftw package (tarballed) in cvs001/fftw/fftw-2.1.3
	Added some general fftw papers in cvs001/fftw

2002-05-18 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	Changed the vector data sturcture through out cepDataset
	some general fixes necessary...
	With Michael and Kristy.

2002-05-20 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	continued changing/fixes for the new vector data sturcture 
	through out cepDataset and some general fixes...

2002-05-20 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	added meeting minutes for weeks 13, 14, 15. Still more to come.

2002-06-13 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	modified week13 minutes to include meeting with Peter Morgan
	modified configure.in to to link fftw libraries
	committed configure after re running autoconf
	
2002-07-23 Kristy Van Der Vlist bobhchicken@bigpond.com
	added buildreport.pl to the meetings directory. This script should be
	used to generate the meeting reports rather than buildreport.sh as it
	fixes the problems associated with the first one. 

2002-08-04 Kristy Van Der Vlist bobhchicken@bigpond.comi
	added final version of cepMatrix to source. Also added updated version
	of cepLs - the residual calcs are to come.

2002-08-16 Daniel Fernandez daniel.fernandez@ise.canberra.edu.au
	added fft template class and test code



cep/source/cepInterpUi.cpp
/*
  * Imp for the Interpolation wxWindows UI implementation
  * Copyright (C) Kristy Van Der Vlist             2002
  *
  * This program is free software; you can redistribute it and/or modify it
  * under the terms of the GNU General Public License as published by the Free
  * Software Foundation; either version 2 of the License, or (at your option)
  * any later version.
  *
  * This program is distributed in the hope that it will be useful, but WITHOUT
  * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
  * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
  * more details.
  *
  * You should have received a copy of the GNU General Public License along
  * with this program; if not, write to the Free Software Foundation, Inc., 675
  * Mass Ave, Cambridge, MA 02139, USA.
*/

#include "cepInterpUi.h"
  
BEGIN_EVENT_TABLE (cepInterpShowRate, wxDialog)
  EVT_BUTTON(CEPBTN_RATE_SUBMIT, cepInterpShowRate::dlgRateOnOK)
  EVT_BUTTON(CEPBTN_RATE_CANCEL, cepInterpShowRate::dlgRateOnQuit)
  EVT_CLOSE( cepInterpShowRate::dlgRateOnQuit)
END_EVENT_TABLE ()
                          
cepInterpShowRate::cepInterpShowRate():
  wxDialog((wxDialog *) NULL, -1, "Specify Sample Rate", wxPoint(120,120), wxSize(200, 200))
{
  m_panel = new wxPanel(this, -1, wxPoint(120,120), wxSize(200,200));

  m_statBox = new wxStaticBox(m_panel, -1, "", wxPoint(15, 30), wxSize(170, 120));

  m_statText1 = new wxStaticText(m_panel, -1, "Please specify the sample rate", wxPoint(5,5), wxSize(190, 20), wxALIGN_CENTRE);
  m_statText2 = new wxStaticText(m_panel, -1, "for the interpolated data:", wxPoint(5,19), wxSize(190, 20), wxALIGN_CENTRE);

  m_statText3 = new wxStaticText(m_panel, -1, "Sample Rate:", wxPoint(25,40), wxSize(100, 20), wxALIGN_LEFT);

  m_tbSample = new wxTextCtrl(m_panel, -1, "0.0", wxPoint(110, 40), wxSize(60, 20));

  m_rbYear = new wxRadioButton(m_panel, -1, "Years", wxPoint(25, 65), wxSize(120, 20), wxRB_GROUP);
  m_rbDays = new wxRadioButton(m_panel, -1, "Month", wxPoint(25, 90));
  m_rbHours = new wxRadioButton(m_panel, -1, "Days", wxPoint(25, 115));

  m_bSubmit = new wxButton(m_panel, CEPBTN_RATE_SUBMIT, "Ok", wxPoint(10,160));
  m_bSubmit->SetDefault();
  m_bCancel = new wxButton(m_panel, CEPBTN_RATE_CANCEL, "Cancel", wxPoint(110,160));

  Center();
  ShowModal();
}

double cepInterpShowRate::getSample()
{
  for(size_t i = 0; i < m_sampleRate.Length(); i ++)
  {
    if(cepIsNumeric(m_sampleRate.GetChar(i)) == false)
    {
      return -2.0;
    }
  }

  return (atof(m_sampleRate.c_str()));
}

cepInterpShowRate::units cepInterpShowRate::getUnits()
{
  return m_sampleUnits;
}
        
void cepInterpShowRate::dlgRateOnQuit(wxCommandEvent& WXUNUSED(event))
{
  //if cancel or quit button pressed 
  m_sampleRate = "-1";
  
  EndModal(1);
  Destroy();
}

void cepInterpShowRate::dlgRateOnOK(wxCommandEvent& WXUNUSED(event))
{
  if(m_rbYear->GetValue() == true)
  {
    m_sampleUnits = years;
  }
  else
  {
    if(m_rbDays->GetValue() == true)
    {
      m_sampleUnits = days;
    }
    else
    {
      if(m_rbHours->GetValue() == true)
      {
        m_sampleUnits = hours;
      }
      else
      {
        m_sampleUnits = unknowen;
      }
    }
  }

  m_sampleRate = m_tbSample->GetValue();
  
  EndModal(0);
  Destroy();  
}

cepInterpUi::cepInterpUi() {}

void cepInterpUi::showSampleRate()
{
  cepInterpShowRate sr;

  m_sampleUnits = sr.getUnits();
  m_sampleRate = sr.getSample();
}

double cepInterpUi::getSampleRate()
{
  return m_sampleRate;
}

cepInterpShowRate::units cepInterpUi::getUnits()
{
  return m_sampleUnits;
}
